tests/test_generate.py

```diff
@@ -1,69 +1,56 @@
 import unittest
+
 import cohere
+
 from utils import get_api_key
 
 API_KEY = get_api_key()
 co = cohere.Client(API_KEY)
 
 
 class TestGenerate(unittest.TestCase):
+
     def test_success(self):
-        prediction = co.generate(
-            model='small',
-            prompt='co:here',
-            max_tokens=1)
+        prediction = co.generate(model='small', prompt='co:here', max_tokens=1)
         self.assertIsInstance(prediction.generations[0].text, str)
         self.assertIsNone(prediction.generations[0].token_likelihoods)
         self.assertEqual(prediction.return_likelihoods, 'NONE')
 
+    def test_success_batched(self):
+        _batch_size = 10
+        predictions = co.batch_generate(model='small', prompts=['co:here'] * _batch_size, max_tokens=1)
```

Comment:
More for my understanding but in a batch_generate workflow, what if one of the prompts is too long? How would that error be returned to the user?

Additionally, how would batch_generate be rate limited? (Since it is a single request from the user)
Do we have a limit on our end on the max number of prompts that a user can specify for a specific prompt?

My concern is that a trial key user can send a batch generate call with 1M generations and it would be treated as 1 request and heavily skew our latencies for all other users

Reply:
batch_generate is a bunch of individual generates under the hood. So rate limiting is the same as doing a bunch of indiv. calls.

For errors, if any of the parallel calls error out the program will error as expected. In a future PR I'm going to improve our error handling in python so that users can have better error handling and the errors are more informative.

Reply:
Awesome. Thank you!